vnda/loaders/productListingPage.ts

```diff
@@ -63,31 +65,32 @@ const searchLoader = async (
   const term = props.term || props.slug || qQueryString ||
     undefined;
 
+  const categoryTagName = props.term || url.pathname.split("/").pop() || "";
+
+  const categoryTag = isSearchPage
+    ? undefined
+    : await api["GET /api/v2/tags/:name"]({ name: categoryTagName }, STALE)
+      .then((res) => res.json()).catch(() => undefined);
+
   const response = await api["GET /api/v2/products/search"]({
     term,
     sort,
     page,
     per_page: count,
-    "tags[]": props.tags,
+    "tags[]": props.tags && props.tags?.length > 0
+      ? props.tags
+      : (categoryTag?.name && props.filterByTags
+        ? [categoryTag?.name]
+        : undefined),
```

Comment:
This seems a little confusing...

create some variables before that call:

```
const initialTags = props.tags && props.tags?.length > 0 ? props.tags : undefined;
const categoryTagsToFilter = (categoryTag?.name && props.filterByTags
        ? [categoryTag?.name]
        : undefined)
```

and use it to filter tags[]:

```suggestion
    "tags[]": initialTags ?? categoryTagsToFilter,
```